### PR TITLE
[ISSUE-46] - Jdbc Realm Definitions.

### PR DESCRIPTION
**Pull request by pedroigor**
Build is failing for this PR. The reason is that I need to figure out how to avoid error due to unavailable capabilities required for our services. The ```JdbcRealmService``` requires the ```DataSource``` API.

I've done some functional testing connecting to a running database to authenticate users. Everything seems to work so far. Specially the capabilities related code.

### CODE REVIEW

**Comment by darranl** [comment]
Depends on this PR https://github.com/wildfly/wildfly/pull/7682

**Comment by darranl** [comment]
Pedro - can you please adjust the commit message so it is in a format GitHub understands: -

https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by pedroigor** [comment]
I've added support to our tests in order to initialize any required capability.

@darranl Good for review and merge.

**Comment by darranl** [comment]
Ok that is a little too far 'Fixes #46 JDBC Realm Backed Defintions' would be a more appropriate commit message ;-)

Although it looks like this may need a rebase.

**Comment by pedroigor** [comment]
The PR title is already 'JDBC Realm Backed Definitions'. So commits inherit the intend of the PR :) Anyway, I've changed the commit message and rebased. From now on, I'll keep this policy in mind.

Thanks !

File: src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+            String algorithm = ElytronExtension.asStringIfDefined(context, ALGORITHM, propertyNode);
+            int password = ElytronExtension.asIntIfDefined(context, PASSWORD, propertyNode);
+            int salt = ElytronExtension.asIntIfDefined(context, SALT, propertyNode);
+            int iterationCount = ElytronExtension.asIntIfDefined(context, ITERATION_COUNT, propertyNode);
+
+            return new PasswordKeyMapper(algorithm, password, salt, iterationCount);
+        }
+    }
+
+    interface PasswordMapperResourceDefinition {
```
Comment (darranl) [inline src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java]: I think 'ResourceDefinition' is the wrong term to use here as that has a distinct definition already - maybe 'PasswordMapperObjectDefinition' and also rename the implementations?

Comment (pedroigor) [inline src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java]: Yeah, makes more sense. Changed names accordingly.

**Pull request merged**